airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/test/util/IntegrationTest.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.test.util
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.cdk.command.ConfigurationJsonObjectBase
+import io.airbyte.cdk.command.DestinationCatalog
+import io.airbyte.cdk.command.DestinationStream
+import io.airbyte.cdk.message.DestinationMessage
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.airbyte.protocol.models.v0.AirbyteStreamStatusTraceMessage
+import io.airbyte.protocol.models.v0.AirbyteStreamStatusTraceMessage.AirbyteStreamStatus
+import io.airbyte.protocol.models.v0.AirbyteTraceMessage
+import io.airbyte.protocol.models.v0.StreamDescriptor
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import kotlin.test.fail
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import org.apache.commons.lang3.RandomStringUtils
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+
+@MicronautTest
+@Execution(ExecutionMode.CONCURRENT)
+// Spotbugs doesn't let you suppress the actual lateinit property,
+// so we have to suppress the entire class.
+// Thanks, spotbugs.
+@SuppressFBWarnings("NP_NONNULL_RETURN_VIOLATION", justification = "Micronaut DI")
+abstract class IntegrationTest(
```

Comment:
@johnny-schmidt / @stephane-airbyte I do want to call this piece out specifically - I copied this pattern from existing tests, but it could (/should) just be some utility class that gets constructed by test classes, instead of an extra layer of inheritance.

I'm basically reserving my next couple weeks to work on testing stuff (and probably won't have time to make big changes b/c OC)... so I'd prefer to just leave that debate for later, so we can at least get things merged + unblock further development? It should be a pretty simple switch, just a bit tedious.

Reply:
I'm fine with this as a foundation. We can talk through some of the implications of this stuff at the next CDK meeting?

Reply:
sgtm!